shared/src/address/payload.rs

```diff
@@ -75,6 +87,21 @@ pub enum Payload {
     Delegated(DelegatedAddress),
 }
 
+#[cfg(feature = "arb")]
+impl quickcheck::Arbitrary for Payload {
+    fn arbitrary(g: &mut quickcheck::Gen) -> Self {
+        let id = Payload::ID(u64::arbitrary(g));
+        let secp = Payload::Secp256k1([u8::arbitrary(g); PAYLOAD_HASH_LEN]);
```

Comment:
This generates a single arbitrary `u8` and then repeats it `PAYLOAD_HASH_LEN` times.

Reply:
Fixed